src/cls-ah.ts

```diff
@@ -97,10 +98,11 @@ class AsyncHooksNamespace implements CLSNamespace {
     EVENT_EMITTER_METHODS.forEach((method) => {
       // TODO(kjin): Presumably also dependent on MS/TS-#15473.
       // tslint:disable:no-any
-      const oldMethod = (ee as any)[method];
-      (ee as any)[method] = function(this: {}, event: string, cb: Func<void>) {
-        return oldMethod.call(this, event, ns.bind(cb));
-      };
+      shimmer.wrap(ee, method, (oldMethod) => {
+        return function(this: {}, event: string, cb: Func<void>) {
+          return oldMethod.call(this, event, ns.bind(cb));
```

Comment:
Nit. AIUI `bind()` is slow. Can this be
```typescript
return oldMethod.call(this, event, (...args: Array<{}>) => cb.call(ns, ...args));
```
?

I'm not actually sure which one is better.

Reply:
The name `bind` is overloaded... but this doesn't refer to `Function#bind`, but rather `CLSNamespace#bind`.

Reply:
Oh, I see. Didn't know about CLS.